shelldio.sh

```diff
@@ -18,6 +18,27 @@ my_stations="$HOME/.shelldio/my_stations.txt"
 
 ### Functions List
 
+validate_csv() {
+	awk 'BEGIN{FS=","}!n{n=NF}n!=NF{failed=1;exit}END{print !failed}' "$1"
```

Comment:
Την τεσταρα και όντως παίζει 👍 

```
$ ./shelldio.sh
Πρόβλημα: Η λίστα σταθμών: /Users/drpaneas/.shelldio/all_stations.txt δεν είναι έγκυρη
Εκτέλεσε shelldio --fresh για να κατεβάσεις τη λίστα εκ νέου
```

Από περιέργεια, θα μπορούσες μήπως αν έχεις τον χρόνο να εξηγήσεις πώς λειτουργεί αυτό το μαγικό one-liner; :D

Reply:
Ναι αμέ!
Οι μεταβλητές στην `awk` ειναι persistent. Οπότε στην πρώτη γραμμή, το `n` που δεν υπαρχει ακομα (`!n`) γινεται initialize με το `NF`. `NF` είναι `Number of Fields in a record`, οπότε κάνω count τα splitted parts της κάθε γραμμής. Split γίνεται στον χαρακτήρα `,`, λόγω του `FS=","` όπου `FS` είναι `Field separator variable`. Για όλες τις γραμμές, τσεκάρω ότι το `NF` δεν είναι διάφορο του `n`. Αν είναι ορίζω τη μεταβλητή `failed` και κάνω exit. Το υπόλοιπο είναι pretty straight forward. `print !failed` :)